src/clients/opensea/index.ts

```diff
@@ -49,31 +55,50 @@ export class Opensea {
       consideration,
     });
 
-    const functionName = transaction.function.split('(')[0];
-    const isMatchAdvancedOrders = functionName === 'matchAdvancedOrders';
-    const isFulfillAdvancedOrder = functionName === 'fulfillAdvancedOrder';
+    const isMatchAdvancedOrders = (
+      transaction: FulfillmentDataResponse['fulfillment_data']['transaction'],
+    ): transaction is MatchAdvancedOrdersTransaction => {
+      return transaction.function.split('(')[0] === 'matchAdvancedOrders';
+    };
+    const isFulfillAdvancedOrder = (
+      transaction: FulfillmentDataResponse['fulfillment_data']['transaction'],
+    ): transaction is FulfillAdvancedOrderTransaction => {
+      return transaction.function.split('(')[0] === 'fulfillAdvancedOrder';
+    };
+
+    let functionArgs: unknown[] = [];
```

Comment:
What if you don't type this? Will it fail? I'd avoid typing with this too, don't like it at all.